packages/bot/src/dashboard/controllers.ts

```diff
@@ -34,3 +41,45 @@ export async function getRepositorySummary(): Promise<RepositorySummary> {
   }
   return Object.values(repoMap);
 }
+
+export async function getViolatedCommits(
+  pageNumber: number = 1
+): Promise<Commit[]> {
+  if (pageNumber < 1) {
+    pageNumber = 1;
+  }
+  const commits = await getCommitPage((pageNumber - 1) * pageSize, pageSize);
+  return commits;
+}
+
+interface CommitCounts {
+  totalCount: bigint;
```

Comment:
It does not seems useful to use `bigint` for total count while using `number` for page count; when we need to use `bigint` for total count, certainly we'll want/need to use it for page count as well. Let's just use plain number as we won't have this number of commits in the foreseeable future.